CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift

```diff
@@ -54,31 +61,61 @@ public class AnalyticsManager {
 
         for property in properties {
             let propertyValue = property.value.analyticsPropertyValue()
-            if let propertyValue = propertyValue as? Bool {
-                eventProperties[property.key.rawValue] = analyticsString(from: propertyValue)
+            eventProperties[property.key.rawValue] = convertPropertyValueToString(propertyValue)
+        }
+
+        mixpanelInstance?.track(event: event.rawValue, properties: eventProperties)
+    }
+
+    public static func trackUserProperties(_ properties: [AnalyticsUserProperty: AnalyticsPropertyValue]) {
+        if mixpanelInstance != nil {
+            var propertiesToTrack: [String: String] = [:]
+
+            for (property, value) in properties {
+                propertiesToTrack[property.rawValue] = convertPropertyValueToString(value)
             }
 
-            if let propertyValue = propertyValue as? String {
-                eventProperties[property.key.rawValue] = propertyValue
+            mixpanelInstance?.people.set(properties: propertiesToTrack)
+        } else {
+            for (property, value) in properties {
+                userProperties[property] = value
             }
+        }
```

Comment:
[clarification]: I don't understand if `trackUserProperties` and `registerSuperPropertiesOnce` are called only once when the SDK is initialized, does it still make sense to have the else statement here? Also this methods are called right after we create a `mixpanelInstance`.
Also, we don't do anything when we track a normal event: `mixpanelInstance?.track(event: event.rawValue, properties: eventProperties)`

Reply:
`trackUserProperties` and `registerSuperPropertiesOnce` can be called before Analytics SDK inits, because some of user properties (returnAssistantEnabled & returnReasonsEnabled) and super properties (client id) are tracked from BankSDK. I made arrays private and all tracking access via funcs. It will allow us in future to not care about Analytics init, we will track everything via same methods

Reply:
@mrkulik true. I forgot the real reason of doing this. Thanks for reminding me this.